src/cosim/timer.cpp

```diff
@@ -72,21 +77,36 @@ class real_time_timer::impl
     std::shared_ptr<real_time_config> config_;
     size_t configHashValue_;
     std::shared_ptr<real_time_metrics> metrics_;
+    std::optional<duration> sampling_period_to_monitor_ = std::nullopt;
+
+    void update_rolling_average_real_time_factor(
+        Time::time_point& currentTime,
+        time_point& currentSimulationTime,
+        const duration& elapsedRealTime)
+    {
+        const auto elapsedSimTime = currentSimulationTime - rtSimulationStartTime_;
+        metrics_->rolling_average_real_time_factor = elapsedSimTime.count() / (1.0 * elapsedRealTime.count());
```

Comment:
`elapsedSimTime` and `elapsedRealTime` are two different `std::chrono::duration` types associated with different clocks. The former is associated with `Time`, which is an alias for `std::chrono::steady_clock` and represents real time, while the latter is associated with `cosim::detail::clock`, which is a clock defined to represent simulation time. Therefore, it might be fragile to use their `count()` values like this, because it assumes that their tick period is the same.

Two options:
1. Insert `std::duration_cast` calls and cast them to a common type with the same tick period.
2. Insert a compile-time check like `static_assert(decltype(elapsedSimTime)::period == decltype(elapsedRealTime)::period);`

I think I prefer the first one if it works, but there may be arguments in favour of the latter too.

Reply:
I see now that this is just copied from old code elsewhere in the file. But then this is a good opportunity to fix it. ;)

Reply:
Added `duration_cast` as well as in `update_rolling_average_real_time_factor`.

Reply:
After adding duration_cast (or some other updates), I see noticeable performance drops. Investigating on it..